disk/datadog_checks/disk/disk.py

```diff
@@ -147,47 +127,94 @@ def collect_metrics_psutil(self):
 
         self.collect_latency_metrics()
 
-    def _exclude_disk_psutil(self, part):
+    def exclude_disk(self, part):
         # skip cd-rom drives with no disk in it; they may raise
         # ENOENT, pop-up a Windows GUI error for a non-ready
         # partition or just hang;
         # and all the other excluded disks
         skip_win = Platform.is_win32() and ('cdrom' in part.opts or part.fstype == '')
         return skip_win or self._exclude_disk(part.device, part.fstype, part.mountpoint)
 
-    def _exclude_disk(self, name, filesystem, mountpoint):
+    def _exclude_disk(self, device, file_system, mount_point):
         """
         Return True for disks we don't want or that match regex in the config file
         """
-        self.log.debug('_exclude_disk: {}, {}, {}'.format(name, filesystem, mountpoint))
+        self.log.debug('_exclude_disk: {}, {}, {}'.format(device, file_system, mount_point))
+
+        if not device or device == 'none':
+            device = None
+
+            # Allow no device if `all_partitions` is true so we can evaluate mount points
+            if not self._all_partitions:
+                return True
 
         # Hack for NFS secure mounts
         # Secure mounts might look like this: '/mypath (deleted)', we should
-        # ignore all the bits not part of the mountpoint name. Take also into
-        # account a space might be in the mountpoint.
-        mountpoint = mountpoint.rsplit(' ', 1)[0]
+        # ignore all the bits not part of the mount point name. Take also into
+        # account a space might be in the mount point.
+        mount_point = mount_point.rsplit(' ', 1)[0]
 
-        name_empty = not name or name == 'none'
+        return (
+            self._blacklist_disk(device, file_system, mount_point)
+            or not self._whitelist_disk(device, file_system, mount_point)
+        )
 
-        # allow empty names if `all_partitions` is `true` so we can evaluate mountpoints
-        if name_empty and not self._all_partitions:
+    def _whitelist_disk(self, device, file_system, mount_point):
+        if not self._whitelist_file_system(file_system):
```

Comment:
What do you think about the following ? A disk is whitelisted iff file_system and device and mount point are whitelisted, so I feel like this is both clearer and shorter.
```suggestion
       return (
           self._whitelist_file_system(file_system) and
           self._whitelist_device(device) and
           self._whitelist_mount_point(mount_point)
       )
```